src/highlight/highlightUtil.ts

```diff
@@ -1,4 +1,61 @@
-import { Annotation, AnnotationHighlight, Type } from '../@types';
+import { Annotation, AnnotationHighlight, Position, Type } from '../@types';
+
+type Shape = Required<DOMRectInit>;
+
+export const getBoundingRect = (shapes: Shape[]): Shape => {
+    let minX = Number.MAX_VALUE;
+    let minY = Number.MAX_VALUE;
+    let maxX = Number.MIN_VALUE;
+    let maxY = Number.MIN_VALUE;
+
+    shapes.forEach(({ height, width, x, y }) => {
+        const x2 = x + width;
+        const y2 = y + height;
+
+        if (x < minX) {
+            minX = x;
+        }
+
+        if (y < minY) {
+            minY = y;
+        }
+
+        if (x2 > maxX) {
+            maxX = x2;
+        }
+
+        if (y2 > maxY) {
+            maxY = y2;
+        }
+    });
+
+    return {
+        x: minX,
+        y: minY,
+        width: maxX - minX,
+        height: maxY - minY,
+    };
+};
+
+const centerShape = (shape: Required<DOMRectInit>): Position => {
```

Comment:
Should this use `Shape`? It seems like we should either alias it as Shape in the top-level types (near Position) or use `Required<DOMRectInit>` directly, everywhere. Thoughts?

Reply:
Seems like since it's so pervasive, makes sense to alias it as Shape and use that since `Required<DOMRectInit>` is a bit verbose